components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -230,6 +234,9 @@ private PostAuthnHandlerFlowStatus handleResponseFlow(HttpServletRequest request
                     callDefaultProvisioningHandler(username, context, externalIdPConfig, combinedLocalClaims,
                             stepConfig);
                     handleConsents(request, stepConfig, context.getTenantDomain());
+                    // Trigger event to store last login time after successful authentication.
+                    String emailUsername = localClaimValues.get(EMAIL_ADDRESS_CLAIM);
+                    triggerEvent(username, context, emailUsername);
```

Comment:
why is this coupled to email claims specifically? username can be non-email as well.

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/4612/commits/a4e3a9878fea0d2b572975f81dffafd696fbe36c